packages/swingset-liveslots/src/watchedPromises.js

```diff
@@ -235,13 +235,13 @@ export function makeWatchedPromiseManager({
       } else {
         watchedPromiseTable.init(vpid, harden([[watcher, ...args]]));
 
+        promiseRegistrations.init(vpid, p);
+        pseudoThen(p, vpid);
```

Comment:
```suggestion
        promiseRegistrations.init(vpid, p);

        // pseudoThen registers a settlement callback that will remove this
        // promise from promiseRegistrations and watchedPromiseTable, which
        // should precede `maybeExportPromise` to avoid preventing slotToVal
        // cleanup. This isn't foolproof, but does mitigate
        // https://github.com/Agoric/agoric-sdk/issues/10757 in advance of a
        // proper fix.
        pseudoThen(p, vpid);
```